logic/tool/TypeResolver.java

```diff
@@ -170,10 +178,50 @@ private TypeVariable register(TypeVariable var) {
             }
             resolverRegister.put(var.id(), resolver);
             variableRegister.putIfAbsent(resolver.reference(), var);
-            resolver.addTo(traversal);
+            if (!var.resolvedTypes().isEmpty()) traversal.labels(resolver.id(), var.resolvedTypes());
+            resolver.abstractConstraint().ifPresent(constraint -> registerAbstract(resolver));
+            resolver.is().forEach(constraint -> registerIsType(resolver, constraint));
+            resolver.owns().forEach(constraint -> registerOwns(resolver, constraint));
+            resolver.plays().forEach(constraint -> registerPlays(resolver, constraint));
+            resolver.regex().ifPresent(constraint -> registerRegex(resolver, constraint));
+            resolver.relates().forEach(constraint -> registerRelates(resolver, constraint));
+            resolver.sub().ifPresent(constraint -> registerSub(resolver, constraint));
+            resolver.valueType().ifPresent(constraint -> registerValueType(resolver, constraint));
```

Comment:
two and a half things:
1. shouldn't this be operating over the constraints of `var`, not `resolver`?
1a. if so, do you have any tests that would have caught this mistake?
2. if the order is not important here, I think its less error prone to do
```var.constraints().forEach(constraint -> {
if (constraint.isLabel()) ...
else if (constraint.isAbstract()) ...
```
because this would mean at the end you can throw an exception in the `else`, which means you know if you've accidentally forgotten to handle a constraint, or if we add a new one?

Reply:
3. do we have any tests that show why this catches a bug?

Reply:
1. Yes good point. And good point there isn't a test for that...
2. Could do. Doesn't feel as pretty somehow. But I see your point about safely catching all the constraints.

Reply:
So it turns out the tests weren't checking the anonymous variables. I've fixed that, and now also put a test in to ensure that the `resolve()` function can further prune the `resolvedLabels` on a `role`.